File: SrdDatabase/ClientApp/src/models/sacco/loanInstallment.ts
import { PagedParameters, PagedResults } from "../shared";

export interface LoanInstallment {
    id?: number;
    loanId?: number;
    memberId?: number;
    member?: string;
    date?: Date;
    amount?: number;
    receiptNumber?: number;
}

export interface LoanInstallmentToSend extends Omit<LoanInstallment, "date"> {
    date?: string;
}

export interface LoanInstallmentDetails {
    loanInstallment: LoanInstallment;
}

export interface LoanInstallmentParameters extends PagedParameters {
    memberId?: number;
    startDate?: Date;
    endDate?: Date;
    receiptNumber?: number;
}

export interface LoanInstallmentParametersToSend extends Omit<LoanInstallmentParameters, "startDate" | "endDate"> {
    startDate?: string;
    endDate?: string;
};

export interface LoanInstallmentResults extends PagedResults {
    loanInstallments: LoanInstallment[];
}

File: SrdDatabase/Models/Sacco/LoanInstallments/LoanInstallment.cs
using System;

namespace SrdDatabase.Models.Sacco.LoanInstallments
{
    public class LoanInstallment
    {
        public int Id { get; }

        public int LoanId { get; }

        public int MemberId { get; }

        public string Member { get; }

        public DateTime Date { get; }

        public int Amount { get; }

        public int? ReceiptNumber { get; }

        public LoanInstallment(
            int id,
            int loanId,
            int memberId,
            string member,
            DateTime date,
            int amount,
            int? receiptNumber)
        {
            Id = id;
            LoanId = loanId;
            MemberId = memberId;
            Member = member;
            Date = date;
            Amount = amount;
            ReceiptNumber = receiptNumber;
        }

        // for Dapper
        public LoanInstallment()
        {
        }
    }
}
